lib/tesla/adapter/finch.ex

```diff
@@ -153,5 +152,19 @@ if Code.ensure_loaded?(Finch) do
           {:error, :timeout}
       end
     end
+
+    defp handle_stream_response({:ok, _acc}, ref, owner) do
+      send(owner, {ref, :eof})
+    end
+
+    if Application.spec(:finch, :vsn) >= ~c"0.20.0" do
+      defp handle_stream_response({:error, error, _acc}, ref, owner) do
+        send(owner, {ref, {:error, error}})
+      end
+    else
+      defp handle_stream_response({:error, error}, ref, owner) do
+        send(owner, {ref, {:error, error}})
+      end
```

Comment:
coukd we pattern match in both returns without checking `finch` 🤔 so which ever one matches that is the return type of thing

Reply:
aaahh yeah yeah 😭  what do you think? I don't mind the explicit ignore 🤔 

Anyway, no strong preferences, except the safest one